math/vc_aos/include/algebra/math/impl/vc_aos_vector.hpp

```diff
@@ -88,11 +85,48 @@ ALGEBRA_HOST_DEVICE constexpr auto normalize(const vector_t &v) {
 /// @param v the input vector
 template <algebra::concepts::vc_aos_vector vector_t>
 ALGEBRA_HOST_DEVICE constexpr auto eta(const vector_t &v) noexcept {
-
   return algebra::math::atanh(v[2] / norm(v));
 }
 
-/// Cross product between two input vectors - 3 Dim
+/// Cross product between two input vectors - 3 Dim (single precision)
+///
+/// @tparam vector_t generic input vector type
+///
+/// @param a the first input vector
+/// @param b the second input vector
+///
+/// @return a vector representing the cross product
+template <algebra::concepts::vc_aos_vector vector_t1,
+          algebra::concepts::vc_aos_vector vector_t2>
+requires(std::same_as<algebra::traits::value_t<vector_t1>, float>
+             &&std::same_as<algebra::traits::value_t<vector_t2>, float>)
```

Comment:
Why do we need two separate use cases for float and double?

Reply:
The float implementation compiles into what looks like inefficient code for double precision - as far as I can tell. It seems to move things from avx registers to sse registers, then doing the rotation, then back etc. The implementation with fma looks much better on double precision vectors (and seems to be about 2-3 times faster in the benchmark)

Reply:
Very strange, but I'm very happy to accept this as fact.